oracle/pkg/updater/updater.go

```diff
@@ -24,6 +25,11 @@ import (
 
 type SettlementType string
 
+type TxMetadata struct {
+	PosInBlock int
```

Comment:
nit: this could be `uint` if it's purely used as position index

Reply:
We were already using int as the primary unit. So I would avoid changing it. See [here](https://github.com/primev/mev-commit/pull/161/files#diff-a72bfa02be64c94fbaff07d9ab73a24df390425c69873595cecbaf3cfa5058ccL95).